### PR TITLE
refactor `blas/ext/base/dcusumors` to follow current project conventions

### LINKED ISSUE #1474: [RFC]: refactor `blas/ext/base/dcusumors` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dcusumors`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dcusumors) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by itskdhere**
Resolves #1474 .

## Description

> What is the purpose of this pull request?

This pull request:

-   refactors and updates the existing implementation of `blas/ext/base/dcusumors` to follow the current project conventions.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-  fixes #1474
-  progresses #1152

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dcusumors/manifest.json [hunk 1]
```diff
@@ -1,40 +1,77 @@
...
+        "./include"
+      ],
+      "libraries": [
+        "-lm"
+      ],
+      "libpath": [],
+      "dependencies": []
+    }
+  ]
+}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumors/manifest.json]: add a line at end of file

File: lib/node_modules/@stdlib/blas/ext/base/dcusumors/lib/ndarray.native.js [hunk 2]
```diff
@@ -41,26 +42,20 @@ var addon = require( './dcusumors.native.js' );
...
-	}
-	if ( strideY < 0 ) {
-		offsetY += (N-1) * strideY;
-	}
-	viewX = new Float64Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offsetX), x.length-offsetX ); // eslint-disable-line max-len
-	viewY = new Float64Array( y.buffer, y.byteOffset+(y.BYTES_PER_ELEMENT*offsetY), y.length-offsetY ); // eslint-disable-line max-len
+	offsetX = minViewBufferIndex( N, strideX, offsetX );
+	viewX = offsetView( x, offsetX );
+	offsetY = minViewBufferIndex( N, strideY, offsetY );
+	viewY = offsetView( y, offsetY );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumors/lib/ndarray.native.js]:
```suggestion
	offsetX = minViewBufferIndex( N, strideX, offsetX );
	offsetY = minViewBufferIndex( N, strideY, offsetY );

	viewX = offsetView( x, offsetX );
	viewY = offsetView( y, offsetY );
```

File: lib/node_modules/@stdlib/blas/ext/base/dcusumors/src/addon.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	STDLIB_NAPI_ARGV_INT64( env, sum, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, strideY, argv, 5 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, X, N, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, Y, N, strideY, argv, 4 );
+
+	napi_value v;
+	napi_status status = napi_create_double( env, stdlib_strided_dcusumors( N, sum, X, strideX, Y, strideY ), &v );
+	assert( status == napi_ok );
+	return v;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumors/src/addon.c]: Please refer `addon.cpp`, it returns a nullptr so you'll have to change accordingly

File: lib/node_modules/@stdlib/blas/ext/base/dcusumors/src/addon.c [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+ // Register a Node-API module:
+STDLIB_NAPI_MODULE_EXPORT_FCN( addon );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumors/src/addon.c]: add new line at end of file

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Left few comments, thanks for working on this!

**Comment by itskdhere** [comment]
@Pranavchiku I've applied suggested changes. please take a look.

**Review by Planeshifter — APPROVED** [review]
LGTM! Will merge after CI has passed.

File: lib/node_modules/@stdlib/blas/ext/base/dcusumors/manifest.json [hunk 5]
```diff
@@ -1,40 +1,77 @@
...
+      ],
+      "include": [
+        "./include"
+      ],
+      "libraries": [
+        "-lm"
+      ],
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/blas/ext/base/dcusumors",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dcusumors/manifest.json]:
```suggestion
```

**Pull request merged**